service/controller/v2/resource/chart/create.go

```diff
@@ -59,8 +59,7 @@ func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange inte
 		//
 		err = r.helmClient.InstallFromTarball(tarballPath, ns,
 			helm.ReleaseName(chartState.ReleaseName),
-			helm.ValueOverrides(values),
-			helm.InstallWait(true))
+			helm.ValueOverrides(values))
```

Comment:
Maybe an unlikely scenario but what will happen if the install takes longer than the reconciliation period? Will we tell helm to install again even though the last install is still going on?

Reply:
Nope, we return a non-empty create change only if the name of the chart in current state is empty or the name of the chart in the desired state doesn't match the name in the current state, see https://github.com/giantswarm/chart-operator/blob/master/service/controller/v2/resource/chart/create.go#L89.

After sending a request to install tiller will create a release, initially in `PENDING_INSTALL` state. So, in the next reconciliation loop, when we query the current state of the chart here https://github.com/giantswarm/chart-operator/blob/master/service/controller/v2/resource/chart/current.go#L24 the release will be found, either yet in `PENDING_INSTALL` or already in `DEPLLOYED` or `FAILED` state. In any case, the name won't be empty, will match the name of the desired state and the create change will be empty.

Reply:
Nice, thanks for the explanation.